Add MIGRATION-GUIDE.md file (#188)
splitio/react-client: MIGRATION-GUIDE.md

    </SplitFactory>
  );
};
```

or

```javascript
const MyApp = withSplitFactory(mySplitConfig)(MyComponent);
```

with:

```javascript
const MyApp = () => {
  return (
    <SplitFactoryProvider config={mySplitConfig}>
      <MyComponent />
    </SplitFactoryProvider>
  );
};
```

and consider that `factory`, `client` and `manager` properties might be `null` until the SDK has emitted some event:

```javascript
const MyComponent = () => {
  // factoryFromContext === factory, clientFromContext === client, and they are null until some SDK event is emitted
  const { factory: factoryFromContext, client: clientFromContext } = useContext(SplitContext);
  const { factory, client } = useSplitClient();

  // Example to evaluate all your flags when the SDK is ready and re-evaluate on SDK_UPDATE events
  const { manager } = useSplitManager();
  const FEATURE_FLAG_NAMES = manager ? manager.names() : [];
  const { treatments, isReady } = useSplitTreatments({ names: FEATURE_FLAG_NAMES, updateOnSdkUpdate: true }); // updateOnSdkReady is true by default

  return isReady ?
    treatments['feature-flag-1'].treatment === 'on' ?
      <FeatureOn /> :
      <FeatureOff /> :
    <LoadingPage />
}
```

> # Migrating to get React SDK v1.10.0 improvements: Replacing the deprecated `useClient`, `useTreatments`, and `useManager` hooks with the new `useSplitClient`, `useSplitTreatments`, and `useSplitManager` hooks

Reviewer: ah, sorry Emi!
😳 do you think this title (I suggested) is too long? 😔

What would you say about a shorter version, matching the first header:
```suggestion
# Migrating to get React SDK v1.10.0 improvements: Replacing the deprecated `useClient`, `useTreatments`, and `useManager` hooks
```

> # React SDK v1.11.0: replace deprecated components with the new `SplitFactoryProvider` component

Author: Thanks Lena. I will apply with a few minor changes: 
- "depreciated" -> "deprecated",
- "`SplitFactory` component and `withSplitFactory` hook" -> "`SplitFactory` and `withSplitFactory` components"

> The `SplitFactory` and `withSplitFactory` components have been deprecated since React SDK v1.11.0, and will be removed on a future major release.

Author: Idem

> We recommend migrating to the new `SplitFactoryProvider` component instead. This component is a revised version of `SplitFactory` that properly handles SDK side effects (factory creation and destruction) within the React component lifecycle, resolving memory leak issues in React development mode, strict mode and server-side rendering, and also ensuring that the SDK is updated if `config` or `factory` props change.

Author: Will apply the change, but with the 1st item as part of the initial paragraph, since this is not a benefit, but rather an implementation details.

Notable changes to consider when migrating:
 - `SplitFactoryProvider` utilizes the React Hooks API, requiring React 16.8.0 or later, while `SplitFactory` is compatible with React 16.3.0 or later.

> 

Author: That is not exactly the gotcha that happened in the issue. But yes, if they have a combination of multiple SplitFactory, withSplitFactory and the new SplitFactoryProvider, they might have some issues with multiple factories. But that is explained in the docs, and also it is not directly related to the migration itself. 

So I prefer to not mention it, but add a migration example, as we have in the other section:

` To migrate your existing code, replace: <OLD CODE EXAMPLE> with <NEW CODE EXAMPLE>`

 - Updating the `config` prop in `SplitFactoryProvider` reinitializes the SDK with the new configuration, while `SplitFactory` does not reinitialize the SDK. It is recommended to pass a reference to the configuration object (e.g., via a global variable, `useState`, or `useMemo`) rather than a new instance on each render, to avoid unnecessary reinitializations.

 - Updating the `factory` prop in `SplitFactoryProvider` replaces the current SDK instance, unlike `SplitFactory` where it is ignored.

# React SDK v1.10.0: replace deprecated hooks with new ones

The `useClient`, `useTreatments` and `useManager` hooks have been deprecated since React SDK v1.10.0, and will be removed on a future major release.

We recommend migrating to the new versions `useSplitClient`, `useSplitTreatments` and `useSplitManager` respectively, which provide a more flexible API:

> - They accept extra optional parameters, `updateOnSdkReady`, `updateOnSdkReadyFromCache`, `updateOnSdkTimedout` and `updateOnSdkUpdate`, which enable to control when the hook updates the component. For example, you can set `updateOnSdkReady` to `true`, which is `false` by default, to update the component when an `SDK_UPDATE` event is emitted. This is useful when you want to avoid unnecessary re-renders of your components.

Author: Good catch.

The example was actually "`updateOnSdkUpdate` to `true`, which is `false` by default, to update the component when an `SDK_UPDATE`"

- They return an object containing the SDK status properties. These properties are described in the ['Subscribe to events and changes' section](https://help.split.io/hc/en-us/articles/360038825091-React-SDK#subscribe-to-events-and-changes) and enable conditional rendering of components based on the SDK status, eliminating the need to access the Split context or using the client's `ready` promise or event listeners. For example, you can show a loading spinner while the SDK is not ready, and use the `treatments` result to render the variants of your App once the SDK is ready.

> 

Reviewer: ```suggestion

The new hooks accept the same parameters as their depreciated counterparts, but return an object containing the SDK status properties, as shown below:

```

Author: I am not sure how to include this new paragraph without looking redundant with lines 22 and 24. I will chat with you later.

Reviewer: > I am not sure how to include this new paragraph without looking redundant with lines 22 and 24. I will chat with you later.

Ah, I get it. On first read, it just looked like the code snippet was not introduced... How about something like this:
```suggestion
The usage of the new hooks is shown below:
```
Alternately, we could say, "The initialization of the new hooks is shown below" or something more appropriate. The point is just to briefly introduce the code snippet.

Author: Yes. I like "The usage of the new hooks is shown below:". I will include it

```js
const { client, isReady, isReadyFromCache, hasTimedout, lastUpdate } = useSplitClient();
const { treatments, isReady, isReadyFromCache, hasTimedout, lastUpdate } = useSplitTreatments({ names: ['feature-flag-1'] });
const { manager, isReady, isReadyFromCache, hasTimedout, lastUpdate } = useSplitManager();
```

To refactor your existing code, replace:

```javascript
const client = useClient(optionalSplitKey, optionalTrafficType, optionalAttributes);
const treatments = useTreatments(featureFlagNames, optionalAttributes, optionalSplitKey);
const manager = useManager();
```

with:

```javascript
const { client } = useSplitClient({ splitKey: optionalSplitKey, trafficType: optionalTrafficType, attributes: optionalAttributes });
const { treatments } = useSplitTreatments({ names: featureFlagNames, attributes: optionalAttributes, splitKey: optionalSplitKey });
const { manager } = useSplitManager();
```

> and use the status properties to conditionally render your components. For example, do:

Reviewer: ```suggestion
and use the status properties to conditionally render your components. For example, use the following code:
```
